fix(curriculum): add element to create-a-stack tests

nayanadasgupta opened the pull request:
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42322

Changes:

Changed the pop and peek tests by adding an additional item to the stack. This will ensure that the user's implementation must correctly return the item at the top of the stack.

nayanadasgupta committed:
Adding more than one stack item to the stack tests for #42322

nayanadasgupta committed:
restore test for clear

ShaunSHamilton commented:
Hey there,

Thank you, for this PR! Just wanted to mention, we recommend working off of a new branch (not `main`), otherwise `main` becomes spoilt, once this is merged, and makes creating concurrent PRs more difficult for yourself. Also, we recommend a specific style of PR title and commit message, just to help the reviewing process, and general bookkeeping.

This is outlined in better detail here: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request

There is nothing you need to do to this one, but keep it in mind for future contributions. Thank you

gikf reviewed (dismissed):
Thanks for noticing `pop` method also could be improved in similar way. LGTM :+1: 

I'd consider adding another method call to both tests - to confirm `peek` didn't remove element, and that `pop` did remove.

nayanadasgupta committed:
fix(curriculum): improved peek, pop and clear tests

tests that peek does not remove top element and pop does remove top element and addded an element to the stack for the clear test

nayanadasgupta commented:
> Thanks for noticing `pop` method also could be improved in similar way. LGTM 👍
> 
> I'd consider adding another method call to both tests - to confirm `peek` didn't remove element, and that `pop` did remove.

Thanks! I've modified the tests to check that `peek` didn't remove the top element, and that `pop` did remove the top element.

gikf commented on curriculum/challenges/english/10-coding-interview-prep/data-structures/create-a-stack-class.md:
```diff
@@ -89,8 +90,9 @@ The `pop` method should remove and return the top element of the stack
+    test.push('CS61');
     test.push('CS50');
-    return test.pop() === 'CS50';
+    return test.pop() === 'CS50' && test.pop() == 'CS61';
```
Just a little nitpick - let's make second comparison also strict (`===`) for consistency.

nayanadasgupta replied:
Whoops - that was a silly typo! Added another commit to resolve that.

nayanadasgupta committed:
fix(curriculum): fixed typo in pop test

gikf reviewed (approved):
Looks good! :+1:

ojeytonwilliams reviewed (approved):
LGTM :+1:

ojeytonwilliams commented:
@nayanadasgupta  Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝
